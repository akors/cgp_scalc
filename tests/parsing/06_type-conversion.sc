21 + 91 # exact addition without conversion
21.1 + 91.1 # floating addition without conversion
8 + 4.123 # adding float to exact
2.1112 + 32  # adding exact to float

21 - 91 # exact subtraction without conversion
21.0 - 91.1 # floating subtraction without conversion
8 - 4.123 # subtract float from exact
2.1112 - 32  # subtract exact from float

21 * 3 # exact multiplication without conversion
21.27 * 3.14 # floating multiplication without conversion
8 * 4.123 # multiply float with exact
2.1112 * 3  # multiply exact with float

3 / 4 # exact division
3.333 / 4.444 # floating multiplication without conversion
8 / 4.123 # multiply float with exact
2.1112 / 3  # multiply exact with float

