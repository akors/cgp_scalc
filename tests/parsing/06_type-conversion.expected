112
112.2
12.123
34.1112
-70
-70.1
3.877
-29.8888
63
66.7878
32.984
6.3336
0.75
0.75
1.94033
0.703733
